server.go

```diff
@@ -334,6 +345,14 @@ func (sc *serverConn) handleFrame(strm *Stream, fr *FrameHeader) (err error) {
 
 	switch fr.Type() {
 	case FrameHeaders, FrameContinuation:
+		if strm.headersFinished {
+			return NewGoAwayError(ProtocolError, "stream not open")
+		}
+
+		if fr.Flags().Has(FlagEndHeaders) {
```

Comment:
What do you think about checking if the headersFinished is true or not, if it is, then emit an error. Because maybe a client sends a continuation frame after the EndHeaders, for some reason and that might be treated as an error.
```go
if fr.Flags().Has(FlagEndHeaders) {
  if strm.headersFinished {
    return SomeError(InvalidFrame, "Received a header after the headers closed")
  }

  strm.headersFinished = true
}
```

Reply:
This test is just above this one?

Reply:
Ahh sorry, then it's fine. Oke oke, thanks.